cmake/external_rules.cmake

```diff
@@ -90,6 +92,153 @@ function(download_external_sources)
+      set(CMAKE_SUB_BUILD_OPTIONS ${CMAKE_SUB_BUILD_OPTIONS}
+          --config Release)
+    endif()
+    if(MSVC_RUNTIME_LIBRARY_STATIC)
+      set(CMAKE_SUB_CONFIGURE_OPTIONS ${CMAKE_SUB_CONFIGURE_OPTIONS}
+          -DCMAKE_C_FLAGS_RELEASE="/MT"
+          -DCMAKE_CXX_FLAGS_RELEASE="/MT"
+          -DCMAKE_C_FLAGS_DEBUG="/MTd"
+          -DCMAKE_CXX_FLAGS_DEBUG="/MTd")
+      if (CMAKE_BUILD_TYPE STREQUAL "Debug")
+        set(CMAKE_SUB_CONFIGURE_OPTIONS ${CMAKE_SUB_CONFIGURE_OPTIONS}
+            -DCMAKE_C_FLAGS="/MTd"
+            -DCMAKE_CXX_FLAGS="/MTd"
+            -DCMAKE_MSVC_RUNTIME_LIBRARY=MultiThreadedDebug)
+      else()
+        set(CMAKE_SUB_CONFIGURE_OPTIONS ${CMAKE_SUB_CONFIGURE_OPTIONS}
+            -DCMAKE_C_FLAGS="/MT"
+            -DCMAKE_CXX_FLAGS="/MT"
+            -DCMAKE_MSVC_RUNTIME_LIBRARY=MultiThreaded)
+      endif()
+    else()  # dynamic (DLL) runtime
+      set(CMAKE_SUB_CONFIGURE_OPTIONS ${CMAKE_SUB_CONFIGURE_OPTIONS}
+          -DCMAKE_C_FLAGS_RELEASE="/MD"
+          -DCMAKE_CXX_FLAGS_RELEASE="/MD"
+          -DCMAKE_C_FLAGS_DEBUG="/MDd"
+          -DCMAKE_CXX_FLAGS_DEBUG="/MDd")
+      if (CMAKE_BUILD_TYPE STREQUAL "Debug")
+        set(CMAKE_SUB_CONFIGURE_OPTIONS ${CMAKE_SUB_CONFIGURE_OPTIONS}
+            -DCMAKE_C_FLAGS="/MDd"
+            -DCMAKE_CXX_FLAGS="/MDd"
+            -DCMAKE_MSVC_RUNTIME_LIBRARY=MultiThreadedDebugDLL)
+      else()
+        set(CMAKE_SUB_CONFIGURE_OPTIONS ${CMAKE_SUB_CONFIGURE_OPTIONS}
+            -DCMAKE_C_FLAGS="/MD"
+            -DCMAKE_CXX_FLAGS="/MD"
+            -DCMAKE_MSVC_RUNTIME_LIBRARY=MultiThreadedDLL)
+      endif()
+    endif()
+  else()
+    # Propagate Linux build flags.
+    if("${CMAKE_CXX_FLAGS}" MATCHES "-D_GLIBCXX_USE_CXX11_ABI=0")
+      set(SUBBUILD_USE_CXX11_ABI 0)
+    else()
+      set(SUBBUILD_USE_CXX11_ABI 1)
+    endif()
+    set(CMAKE_SUB_CONFIGURE_OPTIONS ${CMAKE_SUB_CONFIGURE_OPTIONS}
+          -DCMAKE_C_FLAGS="-D_GLIBCXX_USE_CXX11_ABI=${SUBBUILD_USE_CXX11_ABI}"
+          -DCMAKE_CXX_FLAGS="-D_GLIBCXX_USE_CXX11_ABI=${SUBBUILD_USE_CXX11_ABI}")
+    if(CMAKE_TOOLCHAIN_FILE)
+      set(CMAKE_SUB_CONFIGURE_OPTIONS ${CMAKE_SUB_CONFIGURE_OPTIONS}
+          -DCMAKE_TOOLCHAIN_FILE=${CMAKE_TOOLCHAIN_FILE})
+    endif()
+  endif()
+  message("Sub-configure options: ${CMAKE_SUB_CONFIGURE_OPTIONS}")
+  message("Sub-build options: ${CMAKE_SUB_BUILD_OPTIONS}")
+
+  if(NOT ANDROID AND NOT IOS)
+    execute_process(
+      COMMAND ${ENV_COMMAND} cmake -DOPENSSL_NO_ASM=TRUE ${CMAKE_SUB_CONFIGURE_OPTIONS} ../boringssl/src
+      WORKING_DIRECTORY ${PROJECT_BINARY_DIR}/external/src/boringssl-build
+      RESULT_VARIABLE boringssl_configure_status
+    )
+    if (boringssl_configure_status AND NOT boringssl_configure_status EQUAL 0)
+      message(FATAL_ERROR "BoringSSL configure failed: ${boringssl_configure_status}")
+    endif()
+  
+    # Run builds in parallel if we know how
+    if(CMAKE_GENERATOR STREQUAL "Unix Makefiles")
+      set(cmake_build_args -j)
```

Comment:
I think this flag works with other generators too. 
https://cmake.org/cmake/help/latest/release/3.12.html#command-line

Reply:
I'm just copying what we do in the download function above.

Reply:
I wonder why we restrict it to only one generator. Anthony might have more input on this but its not a big deal for now.